app/src/main/java/com/example/vybe/AddEdit/AddEditVibeEventActivity.java

```diff
@@ -39,7 +39,7 @@
  * This Activity displays the screen for a user to add a vibe event, or
  * edit an existing vibe event by adding or modifying the different vibe attributes
  */
-public class AddEditVibeEventActivity extends AppCompatActivity implements SocialSituationFieldFragment.OnSocStnSelectedListener, ImageFieldFragment.OnImageSelectedListener, VibeCarouselDialogFragment.OnVibeSelectedListener, LocationSelectionDialog.OnLocationSelectedListener, MapFragment.OnMapFragmentReadyListener {
+public class AddEditVibeEventActivity extends AppCompatActivity implements SocSitFieldFragment.OnSocSitSelectedListener, ImageFieldFragment.OnImageSelectedListener, VibeCarouselDialogFragment.OnVibeSelectedListener, LocationSelectionDialog.OnLocationSelectedListener, MapFragment.OnMapFragmentReadyListener {
```

Comment:
this makes me uncomfortable why is there so many interfaces being implemented???

Reply:
Haha I get what you mean but its for communicating with fragments like we did in our assignment. For example when you select a socsit in the SocSitFieldFragment, the `onSocSitSelected` method of `AddEditVibeActivity` is called.

Reply:
Sure. But I think we should talk to Ali if he has a suggestion as to what we could do.